cpp/src/strings/regex/regcomp.cpp

```diff
@@ -216,8 +216,8 @@ class regex_parser {
     }
 
     /* sort on span start */
-    for (int p = 0; p < cls.size(); p += 2)
-      for (int np = p + 2; np < cls.size(); np += 2)
+    for (size_t p = 0; p < cls.size(); p += 2)
```

Comment:
Should be `std::size_t` everywhere.

```suggestion
    for (std::size_t p = 0; p < cls.size(); p += 2)
```